sources/packages/backend/libs/services/src/queue/queue.service.ts

```diff
@@ -37,4 +42,32 @@ export class QueueService {
       dashboardReadonly: queue.queueConfiguration.dashboardReadonly,
     }));
   }
+
+  /**
+   * Get queue configuration
+   * @param name queue name
+   * @returns queue configuration.
+   */
+  async getQueueConfiguration(name: QueueNames): Promise<Bull.JobOptions> {
+    const queues = await this.getAllQueueConfigurations();
+    const [queueConfig] = queues.filter((queue) => queue.queueName === name);
+    return {
+      attempts: queueConfig.queueConfiguration.retry,
+      backoff: queueConfig.queueConfiguration.retryInterval,
+      repeat: {
+        cron: queueConfig.queueConfiguration.cron,
+      },
+    };
+  }
+
+  /**
+   * Get queue clean up period.
+   * @param name queue name
+   * @returns queue clean up period.
+   */
+  async getQueueCleanUpPeriod(name: QueueNames): Promise<number | undefined> {
```

Comment:
I would suggest using the argument as queueName. This service may grow further and take other names like job name or something possibly.